server/src/main/java/org/elasticsearch/action/support/RetryableAction.java

```diff
@@ -59,17 +60,47 @@ public RetryableAction(
         TimeValue timeoutValue,
         ActionListener<Response> listener,
         String executor
+    ) {
+        this(logger, threadPool, initialDelay, TimeValue.timeValueMillis(Integer.MAX_VALUE), timeoutValue, listener, executor);
```

Comment:
Can we use `TimeValue.MAX_VALUE` instead of the `Integer.MAX_VALUE`, which seems a bit odd? Or simply remove this default and require it to be explicit by callers.